examples/src/examples/graphics/lines.tsx

```diff
@@ -0,0 +1,198 @@
+
+                    // add line connecting points along z axis
+                    if (x > 1) {
+                        positions.push(pt1.x, pt1.y, pt1.z, pt2.x, pt2.y, pt2.z);
+                        colors.push(c1.r, c1.g, c1.b, c1.a, c2.r, c2.g, c2.b, c2.a);
+                    }
+
+                    // add line connecting points along x axis
+                    if (z > 1) {
+                        positions.push(pt1.x, pt1.y, pt1.z, pt3.x, pt3.y, pt3.z);
+                        colors.push(c1.r, c1.g, c1.b, c1.a, c3.r, c3.g, c3.b, c3.a);
+                    }
+                }
+            }
+
+            // submit the generated arrays of lines and colors for rendering
+            app.drawLineArrays(positions, colors);
+
+            // array of Vec3 and Color classes for different way to render lines
+            const grayLinePositions = [];
+            const grayLineColors = [];
+
+            // handle the array of meshes
+            for (let i = 0; i < numMeshes; i++) {
+
+                // move them equally spaced out around in the circle
+                const offset = i * Math.PI * 2 / numMeshes;
+                const entity = meshes[i];
+                entity.setLocalPosition(
+                    30 + 20 * Math.sin(time * 0.2 + offset),
+                    5 + 2 * Math.sin(time + 3 * i / numMeshes),
+                    30 + 20 * Math.cos(time * 0.2 + offset)
+                );
+
+                // add up bounds of all the meshes
+                const thisBounds = entity.render.meshInstances[0].aabb;
+                if (i === 0) {
+                    bounds.copy(thisBounds);
+                } else {
+                    bounds.add(thisBounds);
+                }
+
+                // half of them uses depth testing, the others do not, and so lines show through the mesh
+                const depthTest = i < 0.5 * numMeshes;
+
+                // half of them are rendered in immediate layer, the other half in world layer
+                const layer = i < 0.5 * numMeshes ? immediateLayer : worldLayer;
+
+                // rotate the meshes
+                entity.rotate((i + 1) * dt, 4 * (i + 1) * dt, 6 * (i + 1) * dt);
+
+                // render wiframe sphere around the objects
+                app.drawWireSphere(entity.getLocalPosition(), 3, pc.Color.YELLOW, 30, depthTest, layer);
+
+                // draw a single magenta line from this mesh to the next mesh
+                const nextEntity = meshes[(i + 1) % meshes.length];
+                app.drawLine(entity.getPosition(), nextEntity.getPosition(), pc.Color.MAGENTA);
+
+                // store positions and colors of lines connecting objects to a center point
+                grayLinePositions.push(entity.getPosition(), new pc.Vec3(0, 10, 0));
+                grayLineColors.push(pc.Color.GRAY, pc.Color.GRAY);
+            }
+
+            // render all gray lines
+            app.drawLines(grayLinePositions, grayLineColors);
```

Comment:
Where is the `drawlines` function? I can't see it in the PR

Reply:
application.js line 1739

Reply:
Ah, GitHub collapsed the diff 😓